core/mempool/src/pool_task.rs

```diff
@@ -176,4 +183,73 @@ pub fn spawn_pool(
         future::ok(())
     });
     tokio::spawn(task);
+
+    // Receive transaction gossips
+    let pool5 = pool.clone();
+    let task = inc_tx_gossip_rx.for_each(move |tx_gossip| {
+        // TODO: verify signature
+        if let Err(e) = pool5.add_payload_with_author(tx_gossip.payload, tx_gossip.sender_id) {
+            warn!(target: "pool", "Failed to add payload from tx gossip: {}", e);
+        }
+
+        future::ok(())
+    });
+    tokio::spawn(task);
+
+    let pool6 = pool.clone();
+    let task = Interval::new_interval(gossip_tx_period)
+        .for_each(move |_| {
+            let my_authority_id = match *pool6.authority_id.read().expect(POISONED_LOCK_ERR) {
+                Some(x) => { x },
+                None => { return future::ok(()) },
+            };
+            for their_authority_id in 0..pool6.num_authorities.read().expect(POISONED_LOCK_ERR).unwrap_or(0) {
```

Comment:
you probably should not send to all authorities all the time? the point of gossip is to send to subset from time to time IMO.